AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/StorageEngineAdapter+SQLite.swift

```diff
@@ -20,12 +20,16 @@ final class SQLiteStorageEngineAdapter: StorageEngineAdapter {
         guard let documentsPath = getDocumentPath() else {
             preconditionFailure("Could not create the database. The `.documentDirectory` is invalid")
```

Comment:
Out of curiosity, why do we preconditionFailure here instead of throwing? (I know this isn't touched in this PR, just wondering while I'm reviewing)

Reply:
Seems like the author wanted to completely halt the execution of the program since this seems like an unrecoverable error that we would have no control over?

If our interface changed to allow users to specify what path to save the db in, then maybe throwing would be more appropriate?